Solo_Client/SoloRestServer.cs

```diff
@@ -0,0 +1,200 @@
+        public async Task Action(IHttpContext context)
+        {
+            // Parse action inputs
+            string action_handle = context.Request.QueryString["action_handle"];
+            Dictionary<string, string> args = JsonConvert.DeserializeObject<Dictionary<string, string>>(context.Request.QueryString["action_vars"]);
+            Dictionary<string, string> action_response;
+
+            // Inject Dependencies
+            string state = _server.Locals.GetAs<string>("state");
+            SoloClient client = _server.Locals.GetAs<SoloClient>("client");
+            string executablePath = _server.Locals.GetAs<string>("executablePath");
+            string tipsFilePath = _server.Locals.GetAs<string>("tipsFilePath");
+            string instrument_status;
+
+            if (state == ModuleStatus.BUSY)
+            {
+                action_response = UtilityFunctions.ActionResponse(StepStatus.FAILED, "", "Module is Busy");
+                await context.Response.SendResponseAsync(JsonConvert.SerializeObject(action_response));
+            }
+
+            // If Module isn't busy, try to run the action
+            try
+            {
+                _server.Locals.TryUpdate("state", ModuleStatus.BUSY, _server.Locals.GetAs<string>("state"));
+                switch (action_handle)
+                {
+                    case "run_protocol":
+                        // check if SOLOSoft already running
+                        Process[] processes = Process.GetProcessesByName(Path.GetFileNameWithoutExtension(executablePath));
+
+                        // If SOLOSoft not already running, open it 
+                        if (processes.Length == 0)
+                        {
+                            Console.Out.WriteLine("SOLOSoft not open");
+                            Process.Start(executablePath);
+                            Thread.Sleep(6000);  // wait for program to open
+                            SendKeys.SendWait("{ENTER}");  //press enter to bring up login 
+                            Thread.Sleep(1000); //wait for login window to appear
+                            SendKeys.SendWait("{ENTER}");  //press enter to bypass login screen
+                            Console.Out.WriteLine("SOLOSoft opened");
+                            Console.Out.WriteLine(client.RunCommand("CLOSEALLFILES"));  // ensure no other SOLO protocols open
+                            Console.Out.WriteLine("Closed all files");
+                        }
+                        else
+                        {
+                            Console.Out.WriteLine("SOLOSoft already open");
+                        }
+                        //collect protocol details and save to temp file 
+                        string[] hso_contents = args["hso_contents"].Split('\n');
+                        string hso_basename = args["hso_basename"];
+                        string temp_file_path = "C:\\labautomation\\instructions_wei\\" + args["hso_basename"];
```

Comment:
Probably want to move this to a cli parameter at some point, but I don't think it's urgent tbh.